explanation/journey-of-a-change-to-production.rst

```diff
@@ -0,0 +1,317 @@
+Launchpad has a large database and codebase with strict uptime requirements,
+we want to deploy changes to only one of these at a time and this separation
+helps that as well.
+
+How to test the code and database changes together?
+---------------------------------------------------
+
+It is important to get the database patches submitted and merged to the
+``db-devel`` branch as early as possible. This is because those changes get
+merged to the ``master`` branch only after they are deployed to production
+(the details will be explained in a later question) and it is a pre-requisite
+to get the code changes merged.
+
+While the database patches make their way to production and to the ``master``
+branch, it is okay to copy the database patch from the ``db-devel``
+branch manually and apply it locally in the development environment to
+implement and test the code changes.
+
+How to deploy the database changes to the production environment?
+-----------------------------------------------------------------
+
+The database patches should be submitted to the ``db-devel`` branch and once
+they are approved, they get merged by a landing (aka "merging") bot which
+performs the merge. The trunk branches of Launchpad have ACLs to only allow
+the landing bot to push changes to the trunk branches.
+
+Then the `Launchpad buildbot <http://lpbuildbot.canonical.com>`_, runs the
+full Launchpad test suite on the ``db-devel`` branch and if it passes, the new
+changes get merged to the ``db-stable`` branch.
+
+Once this is done, the changes show up on the `launchpad-db deployable page`_
+and there is a periodic cron job that automatically deploys the new database
+patches, if any, to the `Launchpad staging environment`_.
+
+The patch application times can be checked from the ``staging`` deployment logs.
+Launchpad has strict requirements about the patch application time and it should
+be under 10 seconds.
+
+After confirming the patch application time, the instructions in the
+`fastdowntime deployment process documentation`_ can be used to deploy the
+database patch to production. Our database deployment policy requires
+deploying only one database patch per deployment. If there are multiple patches
+to deploy, they should be deployed separately. Since there are some fragile
+jobs (for example, the PPA and ``ftpmaster`` publishers, librarian garbage
+collector) that shouldn't be interrupted by a database downtime, the deployment
+process ensures that there are no such jobs running before applying the patch
+with a small downtime. At the end of this process, the database changes will get
+merged to the ``master`` branch.
+
+.. _fastdowntime deployment process documentation: https://wiki.canonical.com/InformationInfrastructure/OSA/LaunchpadRollout#Fastdowntime_db_update
+.. _launchpad-db deployable page: https://deployable.ols.canonical.com/project/launchpad-db
+.. _Launchpad staging environment: https://staging.launchpad.net
+
+What is the landing bot?
```

Comment:
I love these small sections explaining the build machinery components and explanations, but this page is also seeming quite large and IMO jumps from the "journey" to these detail explanations too much - would be nice to follow the journey and be able to click to know more about something mentioned there.

Perhaps I'd put these at the end under h1 section or in a separate page? Also, I'd add links to these sections when they are mentioned above  as a list at the top of the page :)